pkg/runtime/setup.go

```diff
@@ -553,6 +544,33 @@ func (s *setup) uninstall(id strfmt.UUID) (rerr error) {
 		return errs.Wrap(err, "Could not unload artifact envdef")
 	}
 
+	// If this is a dynamically imported artifact, tell the ecosystem to remove/undeploy it.
+	if artifact, exists := s.depot.config.Cache[id]; exists && artifact.Namespace != "" {
+		if ecosys := filterEcosystemMatchingNamespace(s.ecosystems, artifact.Namespace); ecosys != nil {
+			installedFiles := []string{}
+			// Find record of our deployment
+			if deployments, ok := s.depot.config.Deployments[id]; ok {
```

Comment:
Please make a public method on the depot to access this. Possibly you could even combine this with the `artifact,extists` logic above seeing as the deployment has the namespace in it. Something like `depot.GetDeployments(id) []Deployment`. Then check the namespace off os those.